logger/umlogger.go

```diff
@@ -0,0 +1,141 @@
+// Package logger provides implementetion of writing Log messages to Graylog, file, and stdout.
+// Supports log levels and destination.
+package logger
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	graylog "gopkg.in/gemnasium/logrus-graylog-hook.v2"
+)
+
+// ErrFailedToConfigureLog is the error returned when configuring failed for some reasons
+var ErrFailedToConfigureLog = errors.New("Failed to init log: failed to configure ")
+
+// StorageConfig contains fileds used in Connect for DSN
+type LogConfig struct {
+	Host       string
+	Port       string
+	PassSecret string
+	PassSHA2   string
+	Output     string
+	Level      string
+	Type       string
+}
+type loggerKeyType int
+
+const loggerKey loggerKeyType = iota
+
+// NullFormatter structure for creating null formatter for logger
+type NullFormatter struct {
+}
+
+// New returns a context that has a logrus logger
+func New(ctx context.Context) context.Context {
+	return context.WithValue(ctx, loggerKey, WithContext(ctx))
+}
+
+// WithContext returns a logrus logger from the context
+func WithContext(ctx context.Context) *log.Logger {
+	if ctx == nil {
+		return log.StandardLogger()
+	}
+
+	if ctxLogger, ok := ctx.Value(loggerKey).(*log.Logger); ok {
+		return ctxLogger
+	}
+
```

Comment:
Do not pass logger in context, use function parameter instead.

Reply:
fixed

Reply:
removed